cli/command/service/rollback_test.go

```diff
@@ -0,0 +1,107 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/cli/internal/test"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/pkg/testutil"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+)
+
+func TestRollback(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		args                 []string
+		serviceUpdateFunc    func(ctx context.Context, serviceID string, version swarm.Version, service swarm.ServiceSpec, options types.ServiceUpdateOptions) (types.ServiceUpdateResponse, error)
+		expectedDockerCliErr string
+	}{
+		{
+			name: "rollback-service",
+			args: []string{"service-id"},
+		},
+		{
+			name: "rollback-service-with-warnings",
+			args: []string{"service-id"},
+			serviceUpdateFunc: func(ctx context.Context, serviceID string, version swarm.Version, service swarm.ServiceSpec, options types.ServiceUpdateOptions) (types.ServiceUpdateResponse, error) {
+				response := types.ServiceUpdateResponse{}
+
+				response.Warnings = []string{
+					"- warning 1",
+					"- warning 2",
+				}
+
+				return response, nil
+			},
+			expectedDockerCliErr: "- warning 1\n- warning 2",
+		},
+	}
+
+	for _, tc := range testCases {
+		err := new(bytes.Buffer)
+		cli := test.NewFakeCli(&fakeClient{
+			serviceUpdateFunc: tc.serviceUpdateFunc,
+		})
+		cli.SetErr(err)
```

Comment:
minor: With the recent change to `NewFakeCli()` you don't need to do this anymore. The default is to set stderr to a bytes.Buffer.

To get the buffer you call `cli.ErrBuffer()`